gtk/gtkfilechooserdialog.c

```diff
@@ -507,6 +509,61 @@ add_button (GtkWidget *button, gpointer data)
 }
 
 static void
+setup_places (GtkFileChooserDialog *dialog)
+{
+  gboolean use_header;
+
+  if (dialog->priv->places_setup)
+    return;
+
+  dialog->priv->places_setup = TRUE;
+
+  g_object_get (dialog, "use-header-bar", &use_header, NULL);
+  if (use_header)
+    {
+      GtkWidget *button;
+      GtkWidget *image;
+      GtkWidget *header;
+      GdkScreen *screen;
+      GtkPlacesSidebar *sidebar;
+      GtkRequisition sidebar_requisition;
+      gboolean sidebar_visible;
+
+      button = gtk_toggle_button_new ();
+      gtk_widget_set_tooltip_text (GTK_WIDGET (button), _("Show other locations"));
+      gtk_widget_set_focus_on_click (button, FALSE);
+      gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
+      image = gtk_image_new_from_icon_name ("folder-symbolic", GTK_ICON_SIZE_MENU);
+      gtk_container_add (GTK_CONTAINER (button), image);
+      gtk_style_context_add_class (gtk_widget_get_style_context (button), "image-button");
+      gtk_style_context_remove_class (gtk_widget_get_style_context (button), "text-button");
+      gtk_widget_show (image);
```

Comment:
these 3 lines would not be needed if you set the image with gtk_button_set_image()